cmake/modules/FindYAMLCPP.cmake

```diff
@@ -0,0 +1,108 @@
+#.rst:
+# FindYAMLCPP
```

Comment:
please note, we already include seastar as a submodule. which check for yaml-cpp as well. but i understand we should not rely on it as Jaeger and crimson are not hard-wired, and they should not. what i suggest is to use the same name for the package to be found -- "yaml-cpp". and expose the same set of variables / target as https://github.com/scylladb/seastar/blob/master/cmake/Findyaml-cpp.cmake does.

to name some ratinales:

1. no need to check yaml-cpp / YAMLCPP twice. as the variables are cached
2. no divergence when searching for yaml-cpp / YAMLCPP in the same project. we **include** seastar. so it would be confusing when developer finds that we are detecting both yaml-cpp and YAMLCPP.
3. avoid the potential risk of compiling and linking against two different yaml-cpp libraries found by these two scripts.